### PR TITLE
Andy Bettisworth

### PR DESCRIPTION
**Pull request by wurde**

### CODE REVIEW

**Review by angeltorgon — COMMENTED** [review]
## Minimum Viable Product Objectives
* [x] Self-Study questions
* [x] `project` table with unique id, name, description, completed columns
* [x] `action` table with unique id, description, notes, completed columns
* [x] Build database and tables with `knex` migrations
* [x] Api endpoints: POST projects, POST actions, GET project by id 

## Stretch Objectives
* [x] Add remaining CRUD operations not required in MVP
* [x] Add `context` support to app

# Project Manager Feedback
## Great
* You completed all MVP objectives! :heavy_check_mark:
* You completed stretch objectives! :heavy_check_mark:
Awesome job Andy!
## Requested Improvements
* Make a react app that consumes your api
## Questions
* How comfortable to you feel setting up a server from scratch?
## Rating {3}
:star: :star: :star:

File: README.md [hunk 1]
```diff
@@ -33,52 +33,70 @@ This will be akin to the Web API that you built in the last sprint, only this ti
...
+A foreign key. This is a key referencing a primary key in another table.
+
 1. What do we need in order to have a _many to many_ relationship between two tables.
 
+We need a third table. This third table will have foreign keys for both tables
+that comprise the many to many relationship. I call this a join table, but it
+could also be called a bridge table. Having a name for something is *extremely
+useful* when trying to communicate, discuss, or reason about that thing.
+
 ## Project Set Up
```
Comment (angeltorgon) [inline README.md:58]: ✔️

File: knexfile.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    },
+    useNullAsDefault: true,
+    migrations: {
+      directory: './server/db/migrations'
+    },
+    seeds: {
+      directory: './server/db/seeds'
+    }
+  }
+};
```
Comment (angeltorgon) [inline knexfile.js:17]: ✔️

File: server/controllers/ActionsController.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      res.status(500).json({ error: { message: 'Server error.' }})
+    }
+  }
+}
+
+/**
+ * Export controller
+ */
+
+module.exports = ActionsController
```
Comment (angeltorgon) [inline server/controllers/ActionsController.js:91]: ✔️

File: server/controllers/ContextsController.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      res.status(500).json({ error: { message: 'Server error.' }})
+    }
+  }
+}
+
+/**
+ * Export controller
+ */
+
+module.exports = ContextsController
```
Comment (angeltorgon) [inline server/controllers/ContextsController.js:88]: ✔️

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,15 @@
+# http://editorconfig.org
```
Comment (angeltorgon) [inline .editorconfig:1]: Andy I love your code! I always learn something new looking at it! Good job!

Comment (wurde) [reply .editorconfig:1]: Thank you Angel. I too like to read other student's implementations of Lambda projects. Everyone has their own way of coding the solution. For this project I tried to bring in the MVC (Model-View-Controller) pattern I use with client code.

**Pull request merged**